src/Cardano/Wallet/Binary.hs

```diff
@@ -431,11 +431,21 @@ decodeTxWitness = do
     _ <- CBOR.decodeListLenCanonicalOf 2
     t <- CBOR.decodeWord8
     case t of
-        0 -> CBOR.decodeTag *> (PublicKeyWitness <$> CBOR.decodeBytes)
+        0 -> do
+            _ <- CBOR.decodeTag
+            bs <- CBOR.decodeBytes
+            case CBOR.deserialiseFromBytes decodePKWitness (BL.fromStrict bs) of
+                Left err -> fail $ show err
+                Right (_, input) -> return input
         1 -> CBOR.decodeTag *> (ScriptWitness <$> CBOR.decodeBytes)
         2 -> CBOR.decodeTag *> (RedeemWitness <$> CBOR.decodeBytes)
         _ -> fail
```

Comment:
As I had problems with it previously, I should check that this fail works, and doesn't produce ~infinite output.